libpod/image.go

```diff
@@ -183,25 +233,114 @@ func (r *Runtime) PushImage(source string, destination string, options CopyOptio
 
 // TagImage adds a tag to the given image
 func (r *Runtime) TagImage(image *storage.Image, tag string) error {
-	return ctr.ErrNotImplemented
+	r.lock.Lock()
+	defer r.lock.Unlock()
+
+	if !r.valid {
+		return fmt.Errorf("runtime is not valid")
+	}
+
+	tags, err := r.store.Names(image.ID)
```

Comment:
How many names do we expect to be associated with an Image?  I think the answer is less that a half dozen and if so this code is fine.  But if we anticipate more we might want to consider using a map here.